From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tom Levy <tomlevy93@gmail.com>
Date: Sat, 4 Sep 2021 19:17:13 +1200
Subject: Add configuration option 'team_members_required'

Default to false, which matches the existing behaviour.

This configuration can be set to true to require the "Team members"
field in the registration form to be filled in.
---
 etc/db-config.yaml                            | 5 +++++
 webapp/src/Form/Type/UserRegistrationType.php | 6 ++++--
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/etc/db-config.yaml b/etc/db-config.yaml
index 8e80cc754..3f4139ea3 100644
--- a/etc/db-config.yaml
+++ b/etc/db-config.yaml
@@ -182,6 +182,11 @@
             default_value: true
             public: true
             description: Show team members on the scoreboard?
+        -   name: team_members_required
+            type: bool
+            default_value: false
+            public: true
+            description: Require team members to be provided during self-registration?
         -   name: show_flags
             type: bool
             default_value: true
diff --git a/webapp/src/Form/Type/UserRegistrationType.php b/webapp/src/Form/Type/UserRegistrationType.php
index 709a6e44a..6a41b20f0 100644
--- a/webapp/src/Form/Type/UserRegistrationType.php
+++ b/webapp/src/Form/Type/UserRegistrationType.php
@@ -180,15 +180,17 @@ class UserRegistrationType extends AbstractType
 
         if ($this->config->get('show_team_members')) {
             $showTeamMembersPublicly = (bool)$this->config->get('show_team_members_publicly');
+            $teamMembersRequired     = (bool)$this->config->get('team_members_required');
             $builder
                 ->add('members', TextareaType::class, [
                     'label' => false,
                     'mapped' => false,
                     'attr' => [
-                        'placeholder' => 'Team members (optional)',
+                        'placeholder' => $teamMembersRequired ? 'Team members' : 'Team members (optional)',
                         'title' => $showTeamMembersPublicly ? 'Displayed on the scoreboard.' : 'Only visible to administrators.',
                     ],
-                    'required' => false,
+                    'required' => $teamMembersRequired,
+                    'constraints' => $teamMembersRequired ? [new NotBlank()] : [],
                     'mapped' => false,
                 ]);
         }
-- 
2.20.1

